posts/atomic-design-bootstrap.md

```diff
@@ -0,0 +1,154 @@
+
+> Yes, but they are already ready, why not reuse them? Especially since many people maintain them?
+
+This is the argument I hear systematically when we talk about this kind of tools "it exists", "why reinvent the wheel?". I will try to answer these remarks beginning with two quotes from __Alla Kholmatova__ book, [Design Systems](http://designsystemsbook.com):
+
+> __Functional Patterns__ are tangible building blocks of the interface. Their purpose is to enable or encourage certain user behaviors.
+
+_page 63_
+
+> Example of __perceptual patterns__ […] include tone of voice, typography, color palette, layouts, illustrations and iconography styles, shapes and textures, spacing, imagery, interactions or animations, and all the specific ways in which those elements are combined and used in an interface.
+
+_page 82_
+
+Thus, for our buttons, it is necessary to reflect before choosing a ready to use component:
+
+* Why do you need a blue button with rounded corners? all states? all sizes? 9 semantic variants proposed?
+* Do you need glued buttons when grouped?
+* How many time need you to change colors? add new variants? remove rounded corners? shadows?
+* What will happen if you need to upgrade Bootstrap? Even if it is a minor release?
+
+You can follow this type of remarks to other [Components](https://getbootstrap.com/docs/4.4/components/buttons) to understand it is not relevant to use them when you need a graphical identity ; because you may take a lot of time to override and maintain them.
+
+Even if I think it's not the Atomic Design principles, some use cases don't need a lot of energy in an important graphical identity. Sometimes the Bootstrap behaviour may be adapted. In this case, it is interesting to __use__ `Components` or [edit parameters](https://getbootstrap.com/docs/4.4/getting-started/theming/#variable-defaults) without overriding them.
+
+# Ok, but, what should I use if I want to use Atomic Design?
+
+Like I said, you must avoid _Bootstrap_ components usage because they may come into conflict with __Functional Patterns__ and __Perceptual Patterns__  of your design.
+
+However, _Bootstrap_ has some useful elements with less impacts on __Functional Patterns__ and  __Perceptual Patterns__.
+
+Let's start with some examples.
+
+## Grid system
+
+![A Bootstrap grid system disposition](https://raw.githubusercontent.com/gnuk/blog-usa/atomic-bootstrap/images/2020/01/atomic-design-bootstrap-grid.png)
+_Bootstrap grid system example from the website_
+
+One of these elements is the [grid system](https://getbootstrap.com/docs/4.4/layout/grid), because it allow a grid simulation with 12 columns an manage some responsive design issues. Even if gutters (space between columns: 30px by default) may have a lot of effects on the building of a grid, they can be removed using `.no-gutters` modifier.
```

Comment:
```suggestion
One of these elements is the [grid system](https://getbootstrap.com/docs/4.4/layout/grid), because it allows a grid simulation with 12 columns to manage some responsive design issues. Even if gutters (space between columns: 30px by default) may have a lot of effects on the building of a grid, they can be removed using `.no-gutters` modifier.
```